lib/mirage/impl/mirage_impl_dns.ml

```diff
@@ -0,0 +1,32 @@
+open Functoria
+open Mirage_impl_time
+open Mirage_impl_mclock
+open Mirage_impl_pclock
+open Mirage_impl_stack
+open Mirage_impl_random
+
+type dns_client = Dns_client
+
+let dns_client = Type.v Dns_client
+
+let generic_dns_client ?(timeout = 5000000000L) nameservers =
```

Comment:
could we not have a default for the timeout, please? It is an optional argument in Dns_client_mirage.Make().connect.

Having the default only once, and in the dns client library, is much nicer than sprinkling around this number.

Reply:
Good question, I'm in favour of a command-line argument - similar to `nameservers`

Reply:
I just added a commit where `timeout` becomes a required key to build a `dns_client` device (note that it can be an optional argument). WDYT?

Reply:
I'm not sure, maybe an optional keyword argument -- i.e. that a user of that device is not required to pass the key? I'm not sure I understand the design space -- if you think this is a good solution, go with it :)

Reply:
Yes, I agree with you so I changed and the user can optionally define a `dns-timeout` argument if he/she wants but it's not needed.